bm_serial_messages.h

```diff
@@ -68,3 +71,41 @@ typedef struct {
   // Flags for self test result
   uint32_t result;
 } __attribute__ ((packed)) bm_serial_self_test_t;
+
+// DFU BELOW HERE.
+typedef struct {
+  // Node id of unit for update
+  uint64_t node_id;
+  // size of image to update
+  uint32_t image_size;
+  // size of chunks to send
+  uint16_t chunk_size;
+  // crc16
+  uint16_t crc16;
+  // major version
+  uint8_t major_ver;
+  // minor version
+  uint8_t minor_ver;
+  // filter for update
+  uint32_t filter_key; 
+  // git hash 
+  uint32_t gitSHA;
+} __attribute__ ((packed)) bm_serial_dfu_start_t;
+
+typedef struct {
+  // offset from image start
+  uint32_t offset;
+  // data length
+  uint32_t length;
+  // data packet
+  uint8_t data[0];
+} __attribute__ ((packed)) bm_serial_dfu_chunk_t;
+
+typedef struct {
+  // Node id of unit reporting test (leave blank for test request)
+  uint64_t node_id;
+  // success of dfu
+  bool success;
+  // Errors for dfu result
+  uint32_t err;
```

Comment:
Should we make this a dfu_status that way we can encode various things like progress, error, completed, etc?

Reply:
SGTM!